src/test/java/se/nackademin/CustomerTest.java

```diff
@@ -0,0 +1,64 @@
+package se.nackademin;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Test;
+
+public class CustomerTest {
+
+    @Test
+
+    public void CustomerSetPhoneNumber() {
+        // Tests that setting a new phonenumber on Customer Object works.
+        Customer cust = new Customer("Björn", "Adolffsson", "Julgränd", "bjorn@jobb.nu", 46738996655L);
+        cust.setPhoneNumber(46709112233L);
+        Long obj1 = cust.getPhoneNumber();
+        Long obj2 = 46709112233L;
+        assertEquals(obj2, obj1);
+    }
+
+    @Test
+
+    public void NoTrailingWhitespace() {
+        // Test that no trailing whitespace occurs in name-field.
+        Customer cust2 = new Customer("Björn", "Adolffsson", "Julgränd", "bjorn@jobb.nu", 46738996655L);
+        String name = cust2.getFirstname();
+        String LastCharInName = name.substring(name.length() - 1);
+        String WhiteSpace = " ";
+        assertNotEquals(LastCharInName, WhiteSpace);
+    }
+
+    @Test
+
+    public void assertEmailMatchesRegex() {
```

Comment:
This test just checks that the test data provided is an email, which doesn't really test the app code. Should instead test the method checkEmail by giving it an invalid email and making sure it throws an error or similar.